test/utils.js

```diff
@@ -1,12 +1,11 @@
 module.exports = {
-  expectThrow: async (promise) => {
-    const errMsg = 'Expected throw not received';
-    try {
-      await promise;
-    } catch (err) {
-      assert(err.toString().includes('invalid opcode'), errMsg);
-      return;
-    }
-    assert.fail(errMsg);
-  },
+  assertRevert: async promise => {
+  try {
+    await promise
+    assert.fail('Expected revert not received')
```

Comment:
This `assert.fail` should go _after_ the catch block, shouldn't it? Otherwise we'll catch this error, and it does indeed contain the "revert" substring. So put this after (outside) the catch block, and put a return at the end of (inside) the catch block.

Reply:
I think after the catch block is equivalent. If `promise` returns as an error, we go immediately to the catch block. If `promise` is successful, then it passes to `assert.fail()`.

Reply:
Does `assert.fail` not emit an error like `assert(false)`?

Reply:
Looking at the API, I think this will work as-written, but I think it's not using the API properly. I think it should be `assert.fail(foo, bar, 'Expected revert not received')`, and if it _were_ that, it would hit the catch with a `revert` substring in the error message.

Reply:
See additional discussion here: https://consensys.slack.com/archives/C037XE8SU/p1519666129000572